src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -112,6 +112,64 @@ public void Write(int field, ReadOnlySpan<byte> value)
         _dataIndex += value.Length;
     }
 
+    public void WriteDynamic(string name, ReadOnlySpan<byte> value, long longValue, double doubleValue)
+    {
+        using var _ = _context.From(name, ByteStringType.Immutable, out var fieldNameStr);
+        Span<byte> tmpBuffer = stackalloc byte[10];
+        int fieldLenLen = VariableSizeEncoding.Write(tmpBuffer, fieldNameStr.Length);
+        long requiredSize = fieldLenLen +
+                            fieldNameStr.Length +
+                            sizeof(IndexEntryFieldType) + 
+                            value.Length + 4 * sizeof(long);
+        
+        if (FreeSpace < requiredSize)
+            UnlikelyGrowAuxiliaryBuffer(requiredSize);
+
+        WriteDynamicFieldName(fieldNameStr);
+        WriteTuple(value, longValue, doubleValue);
+    }
+    
+    /// <summary>
+    /// Writes a textual value to the buffer, as a named value
+    /// </summary>
+    public void WriteDynamic(string name, ReadOnlySpan<byte> value)
+    {
+        using var _ = _context.From(name, ByteStringType.Immutable, out var fieldNameStr);
+        long requiredSize = VariableSizeEncoding.MaximumSizeOf<int>() + // max field len size 
+                            fieldNameStr.Length +
+                            sizeof(IndexEntryFieldType) +
+                            VariableSizeEncoding.MaximumSizeOf<int>() + // max val len size
+                            value.Length;
+        
+        if (FreeSpace < requiredSize)
+            UnlikelyGrowAuxiliaryBuffer(requiredSize);
+        
+        WriteDynamicFieldName(fieldNameStr);
+
+        Span<byte> buffer = Buffer;
+        Unsafe.WriteUnaligned(ref buffer[_dataIndex], IndexEntryFieldType.Raw);
```

Comment:
This is a `IndexEntryFieldType.Simple` not a `IndexEntryFieldType.Raw`

Reply:
Dםמק